eng/pipelines/templates/BuildAndTest.yml

```diff
@@ -23,13 +23,45 @@ parameters:
 steps:
   - script: ${{ parameters.buildScript }}
             -restore -build
-            -pack
             -configuration ${{ parameters.buildConfig }}
             /p:ArchiveTests=true
             /bl:${{ parameters.repoLogPath }}/build.binlog
             $(_OfficialBuildIdArgs)
     displayName: Build
 
+  - ${{ if and(eq(parameters.isWindows, 'true'), ne(parameters.runAsPublic, 'true')) }}:
+    # On windows for internal builds prepare the final signed packages
+    - script: ${{ parameters.buildScript }}
```

Comment:
Do we need this to be a separate step? If not, one alternative would be to combine this and the above, and then use a variable to control the args for whether or not Sign happens

Reply:
Separating it has an additional benefit that we skip `pack` for `non-helix tests` job.

> If not, one alternative would be to combine this and the above, and then use a variable to control the args for whether or not Sign happens

Yeah, we can do this. It will end up a little less clear to read. I will see what it looks like.

Reply:
> Separating it has an additional benefit that we skip pack for non-helix tests job.

Does that really move the needle? From what I've seen (at least locally) pack won't take more than an additional minute or two once build is done.

>Yeah, we can do this. It will end up a little less clear to read.

That's a good point, but I also see other advantages with simplifying it:
- Reduces potential mistakes when variables/parameters are added to t he build (since potentially they won't get set on all the build calls)
- Facilitates local reproing of build issues. Today you have to run 4 build commands in order to mimic what happens in CI, it'd be nice if we can simplify that
- Avoids re-running a bunch of 1ES tasks that get injected each time we call build.cmd as they run each time artifacts are changed.

Reply:
> Avoids re-running a bunch of 1ES tasks that get injected each time we call build.cmd as they run each time artifacts are changed.

oh!

Agreed, working on it now.